packages/gestalt-datepicker/src/DatePicker.js

```diff
@@ -1,37 +1,133 @@
 // @flow strict-local
-import type { ElementRef, Element } from 'react';
-
-import { forwardRef, useEffect, useImperativeHandle, useState, useRef } from 'react';
+import {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useState,
+  useRef,
+  type Element,
+} from 'react';
 import ReactDatePicker, { registerLocale } from 'react-datepicker';
 import classnames from 'classnames';
 import { Icon, Box, Label, Text } from 'gestalt';
 import DatePickerTextField from './DatePickerTextField.js';
 import styles from './DatePicker.css';
 import dateFormat from './dateFormat.js';
-import { type LocaleData } from './LocaleDataTypes.js';
+
+// LocaleData type from https://github.com/date-fns/date-fns/blob/81ab18785146405ca2ae28710cdfbb13a294ec50/src/locale/af/index.js.flow
+// flowlint unclear-type:off
+export type LocaleData = {|
+  code?: string,
+  formatDistance?: (...args: $ReadOnlyArray<any>) => any,
+
+  formatRelative?: (...args: $ReadOnlyArray<any>) => any,
+  localize?: {|
+    ordinalNumber: (...args: $ReadOnlyArray<any>) => any,
+    era: (...args: $ReadOnlyArray<any>) => any,
+    quarter: (...args: $ReadOnlyArray<any>) => any,
+    month: (...args: $ReadOnlyArray<any>) => any,
+    day: (...args: $ReadOnlyArray<any>) => any,
+    dayPeriod: (...args: $ReadOnlyArray<any>) => any,
+  |},
+  formatLong?: {|
+    date: (...args: $ReadOnlyArray<any>) => any,
+    time: (...args: $ReadOnlyArray<any>) => any,
+    dateTime: (...args: $ReadOnlyArray<any>) => any,
+  |},
+  match?: {|
+    ordinalNumber: (...args: $ReadOnlyArray<string>) => any,
+    era: (...args: $ReadOnlyArray<any>) => any,
+    quarter: (...args: $ReadOnlyArray<any>) => any,
+    month: (...args: $ReadOnlyArray<any>) => any,
+    day: (...args: $ReadOnlyArray<any>) => any,
+    dayPeriod: (...args: $ReadOnlyArray<any>) => any,
+  |},
+  options?: {|
+    weekStartsOn?: 0 | 1 | 2 | 3 | 4 | 5 | 6,
+    firstWeekContainsDate?: 1 | 2 | 3 | 4 | 5 | 6 | 7,
+  |},
+|};
+// flowlint unclear-type:error
 
 type Props = {|
+  /**
+   *  See the [disabled example](https://gestalt.pinterest.systems/datepicker#disabled) to learn more.
```

Comment:
Can you add actual descriptions for the props that are missing them? More than just "see example". I've been using these refactors as an excuse to make sure that every prop has a description.

Reply:
DONE